public/New-DbaAgentSchedule.ps1

```diff
@@ -534,6 +534,7 @@ function New-DbaAgentSchedule {
                     if ($PSCmdlet.ShouldProcess($instance, "Adding the schedule $schedule to job $($j.Name)")) {
                         Write-Message -Message "Adding schedule $Schedule to job $($j.Name)" -Level Verbose
                         $j.AddSharedSchedule($jobschedule.Id)
+                        $jobschedule.Refresh()
```

Comment:
Why refresh the object here, when the JobServer gets refreshed on line 528?

Reply:
Adding here would also mean the cache is not getting refreshed properly as well.

Reply:
That seems to be the case. See the [repro in the issue](https://github.com/dataplat/dbatools/issues/8933) to see that this is needed to force the cache to update.

Reply:
That is a byproduct of your repro using Connect, that is not a fix for this command to handle in my view.